app/models/instance.rb

```diff
@@ -15,6 +15,9 @@ def find_tenant_by_host(host)
     end
   end
 
+  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
+  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
```

Comment:
I'm not saying use as many gems as possible. Evaluate them. Sometimes it's trivial, and should not require a dependency. Others are not so. This case it might look trivial but you are implementing an RFC -- not something I'll try to implement in one regex.

Yes, proper naming is a basic requirement. But the problem is that by implementing this separately the Regex is going to become eventually more and more complicated as we add more and more support. At this point, I already cannot completely understand what the regex does, or why it is correct. I know what it's _for_, but not _why_ it works. To be exposed to such a detail while reading the Instance class makes me feel like I'm being presented with too much information. In a separate component, you can break the regex down to different parts, for example. Not something you can do at this level of abstraction.

I don't know about you, but I think the first thing I'd do to find what function does what is to read the Gemfile... that's not saved me all the time, and I still do searches. But what documentation do you propose? Oh, and a debugger helps too. If gems are properly used, the contributor will see the same functions repeat again and again... and the learning cost will be part of the onboarding process. I don't think "more documentation" is the solution to spreading knowledge... because honestly who _wants_ to read documentation, and worse, who _wants_ to write documentation? Do you read documentation when opening a project once? Or do you only start reading when you can't deal with something?